core/src/main/resources/lib/form/checkbox.jelly

```diff
@@ -83,9 +83,9 @@ THE SOFTWARE.
            name="${name}"
            value="${attrs.value}"
            title="${attrs.tooltip}"
-           onclick="${attrs.readonly=='true' ? 'return false;' : attrs.onclick}" id="${attrs.id}" class="${attrs.class} ${attrs.negative!=null ? 'negative' : null} ${attrs.checkUrl!=null?'validated':''}"
+           onclick="${attrs.readonly=='true' ? null : attrs.onclick}" id="${attrs.id}" class="${attrs.class} ${attrs.negative!=null ? 'negative' : null} ${attrs.checkUrl!=null?'validated':''}"
            checkUrl="${attrs.checkUrl}" checkDependsOn="${attrs.checkDependsOn}" json="${attrs.json}"
-           disabled="${readOnlyMode ? 'true' : null}"
+           disabled="${readOnlyMode or attrs.readonly=='true' ? 'true' : null}"
```

Comment:
https://github.com/jenkinsci/jenkins/pull/8420#issuecomment-1962326263 points out this changes the semantics of form submission. Could this cause trouble?

CC @zbynek as comment author

Reply:
Did you check with a screen reader?